compiler/rustc_lint/src/lints.rs

```diff
@@ -556,8 +556,7 @@ pub struct BuiltinUnexpectedCliConfigValue {
 #[diag(lint_supertrait_as_deref_target)]
 pub struct SupertraitAsDerefTarget<'a> {
     pub t: Ty<'a>,
-    pub target_principal: String,
```

Comment:
Now I wonder whether it ever makes sense to store a `String` in a lint diagnostic or whether it would make more sense for the derive to error when it sees a `String`. Might be worth trying out and seeing what happens.